lib/IDE/ExprContextAnalysis.cpp

```diff
@@ -683,6 +679,18 @@ static bool collectPossibleCalleesForUnresolvedMember(
                                               member.Decl))
         candidates.push_back(member);
     }
+  };
+
+  // Get the context of the expression itself.
+  ExprContextInfo contextInfo(&DC, unresolvedMemberExpr);
+  for (auto expectedTy : contextInfo.getPossibleTypes()) {
+    if (!expectedTy->mayHaveMembers())
+      continue;
```

Comment:
Could you move this into `collectMembers()`?

Reply:
Sure, done.